TimeOfFlight/src/TOFEstimators.cc

```diff
@@ -33,442 +43,578 @@
 
-    for( int i=0 ; i< nPFO ; ++i){
+          ReconstructedParticle* pfo = dynamic_cast<ReconstructedParticle*>(  colPFO->getElementAt( i ) ) ;
 
-      ReconstructedParticle* pfo = dynamic_cast<ReconstructedParticle*>(  colPFO->getElementAt( i ) ) ;
+          const ParticleID& tofPID = pidh.getParticleID( pfo , algoID ) ;
 
-      const ParticleID& tofPID = pidh.getParticleID( pfo , algoID ) ;
-      
-      const FloatVec& tofParams = tofPID.getParameters() ;
+          const FloatVec& tofParams = tofPID.getParameters() ;
 
-      streamlog_out( DEBUG ) << " ****  found TOF parameters for pfo w/ size " << tofParams.size() << std::endl ;
+          streamlog_out( DEBUG ) << " ****  found TOF parameters for pfo w/ size " << tofParams.size() << endl ;
 
-      if( !tofParams.empty() ){
+          if( !tofParams.empty() ){
 
-	const double* mom = pfo->getMomentum() ;
-	double momentum = sqrt( mom[0] * mom[0] +  mom[1] * mom[1] +  mom[2] * mom[2] ) ;
+    	const double* mom = pfo->getMomentum() ;
+    	double momentum = sqrt( mom[0] * mom[0] +  mom[1] * mom[1] +  mom[2] * mom[2] ) ;
 
-	double length  =  tofParams[ tof_length  ] ; 
+    	double length  =  tofParams[ tof_length  ] ;
 
-	double beta_fh = ( length / tofParams[ tof_firsthit] ) / 299.8 ;
-	double beta_ch = ( length / tofParams[ tof_closest ] ) / 299.8 ;
+    	double beta_fh = ( length / tofParams[ tof_firsthit] ) / 299.8 ;
+    	double beta_ch = ( length / tofParams[ tof_closest ] ) / 299.8 ;
 
 
-	if( std::abs( pfo->getCharge() )  > 0.5 ) { 
-	  _h[0]->Fill( momentum , beta_fh );
-	  _h[1]->Fill( momentum , beta_ch );
-	} else {
-	  _h[2]->Fill( momentum , beta_fh );
-	  _h[3]->Fill( momentum , beta_ch );
-	}
+    	if( abs( pfo->getCharge() )  > 0.5 ) {
+    	  _h[0]->Fill( momentum , beta_fh );
+    	  _h[1]->Fill( momentum , beta_ch );
+    	} else {
+    	  _h[2]->Fill( momentum , beta_fh );
+    	  _h[3]->Fill( momentum , beta_ch );
+    	}
 
-	if( tofParams[ tof_trk_len  ] > 1.e-3 ){ // if TOF from last tracker hit has been set
+    	if( tofParams[ tof_trk_len  ] > 1.e-3 ){ // if TOF from last tracker hit has been set
 
-	  double beta    = ( tofParams[ tof_trk_len  ] / tofParams[ tof_trkhit  ] ) / 299.8 ;
+    	  double beta    = ( tofParams[ tof_trk_len  ] / tofParams[ tof_trkhit  ] ) / 299.8 ;
 
-	  _h[4]->Fill( momentum , beta );
-	}
+    	  _h[4]->Fill( momentum , beta );
+    	}
 
-      } 
+          }
+        }
+      }
     }
-  }
-  
 }
 
-void TOFEstimators::end(){ 
 
-  gsl_rng_free( _rng );
+void TOFEstimators::end(){ if( _procVersion == "old") gsl_rng_free( _rng ); }
+
+
+Vector3D TOFEstimators::getMomAtCalo(const Track* track){
+    const TrackState* ts = track->getTrackState(TrackState::AtCalorimeter);
+    double phi = ts->getPhi();
+    double d0 = ts->getD0();
+    double z0 = ts->getZ0();
+    double omega = ts->getOmega();
+    double tanL = ts->getTanLambda();
+
+    const Detector& detector = Detector::getInstance();
```

Comment:
```suggestion
    const auto& detector = dd4hep::Detector::getInstance();
```
To "compensate" for the `using dd4hep::Detector` at the top.